internal/chartverifier/checks/helm.go

```diff
@@ -221,52 +217,19 @@ func getImageReferences(chartURI string, vals map[string]interface{}, kubeVersio
 		return nil, err
 	}
 
-	return getImagesFromContent(txt)
+	return getImagesFromContent(txt), nil
 }
 
-func getImagesFromContent(content string) ([]string, error) {
-	imagesMap := make(map[string]bool)
-
-	type ImageRef struct {
-		Ref string `yaml:"image"`
-	}
-
-	r := strings.NewReader(content)
-	reader := bufio.NewReader(r)
-	line, err := getNextLine(reader)
-	for err == nil {
-		var imageRef ImageRef
-		yamlErr := yaml.Unmarshal([]byte(line), &imageRef)
-		if yamlErr == nil {
-			if len(imageRef.Ref) > 0 && !imagesMap[imageRef.Ref] {
-				imagesMap[imageRef.Ref] = true
-			}
-		}
-		line, err = getNextLine(reader)
-		if err == io.EOF {
-			err = nil
-			break
-		}
-	}
-
-	images := make([]string, 0, len(imagesMap))
-	for image := range imagesMap {
-		images = append(images, image)
-	}
-
-	return images, err
-}
-
-func getNextLine(reader *bufio.Reader) (string, error) {
-	nextLine, isPrefix, err := reader.ReadLine()
-	if isPrefix && err == nil {
-		for isPrefix && err == nil {
-			var partLine []byte
-			partLine, isPrefix, err = reader.ReadLine()
-			nextLine = append(nextLine, partLine...)
-		}
+// getImagesFromContent evaluates generated templates from
+// helm and extracts images which are returned in a slice
+func getImagesFromContent(content string) []string {
+	var images []string
+	re := regexp.MustCompile(`\s+image\:\s+(?P<image>.*)`)
```

Comment:
MustCompile makes this a panic if it doesn't, right? If so, can we store this string as a constant and ensure MustCompile works in our unit tests to catch a panic?

Reply:
This error has been handled in the latest commit.